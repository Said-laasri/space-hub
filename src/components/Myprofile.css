.profile {
  justify-content: space-around;
  align-items: center;
  margin: 1rem;
  padding: 1rem;
}

.mission-joined {
  padding: 1rem;
}

.rocket-reserved {
  padding: 1rem;
}

h4 {
  font-size: 2rem;
  font-family: monospace, sans-serif;
  color: rgb(97, 7, 7);
}

h5 {
  font-size: 1rem;
  font-family: fantasy, sans-serif;
}
