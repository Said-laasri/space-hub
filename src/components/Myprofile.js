import { useSelector } from 'react-redux';
import { ListGroup } from 'react-bootstrap';

function Myprofile() {
  const missions = useSelector((state) => state.mission);
  const Missions = missions.filter((mission) => mission.reserved);

  const rockets = useSelector((state) => state.rocket);
  const Rockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div className="profile d-flex">
      <div className="misssion-joined">
        <h4>My Missions</h4>
        <ListGroup>
          {Missions.map((v) => (
            <ListGroup.Item key={v.id}>{v.name}</ListGroup.Item>
          ))}
        </ListGroup>
      </div>
      <div className="rocket-reserved">
        <h4>My Rockets</h4>
        <ListGroup>
          {Rockets.map((v) => (
            <ListGroup.Item key={v.id}>{v.name}</ListGroup.Item>
          ))}
        </ListGroup>
      </div>
    </div>
  );
}

export default Myprofile;
