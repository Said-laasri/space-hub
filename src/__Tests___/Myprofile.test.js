import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configueStore';
import Myprofile from '../components/Myprofile';

it('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Myprofile />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
