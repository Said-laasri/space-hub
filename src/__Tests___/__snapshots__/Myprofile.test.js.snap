// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile d-flex"
>
  <div
    className="misssion-joined"
  >
    <h4>
      Missions
    </h4>
    <div
      className="list-group"
      onKeyDown={[Function]}
    />
  </div>
  <div
    className="rocket-reserved"
  >
    <h4>
      Rockets
    </h4>
    <div
      className="list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
